### ISSUE TITLE
Watchdog Adapter

### ISSUE THREAD

**Issue by klassisch** [ioBroker/AdapterRequests#524]
**What kind of device or service you would like to see an adapter for?**
General units such as sensors or actuators and connections to other systems such as e.g. Homematic CCU

**Is an official API including documentation is available?**
Not needed. Data beeing already available are monitored

**Are other libraries for an integration available?**
Not needed, Watchdog functionality is very genuine and very common as well. I am using three types
* passive and triggered by cyclic data
  * most sensors and systems delivering data to ioBroker in a cyclic manner or at least with a minimum time interval. If this interval will be exceeded, a message will be created
* active triggering a responding device
  * for this purpose one or more oscillators (time bases) triggering other - actively responding devices - devices would be helpful. 
* passive and triggered by offline state
  * some devices ore systems generarte data indicating if they are online or going offline. Of course this situation is more a fluke, moste deveces get depowered and have no chance to indicate this state   

**Is this device already integrated in other Smart Home systems?**
Many systems are using watchdog funtionalities

**Is this device already integrated in homebridge? Might the ham adapter together with the homebridge plugin be sufficient?**
Sorry, I am not familiar with homebridge

**Additional context**
@paul53 have useful hints https://forum.iobroker.net/post/110978 . An adapter developer knows that all already, of course :-)

After you created the issue vote for yourself in the first post of the issue using the "+1"/"Thumbs up" button
Yes, I will try to do

**Comment by Xenon-s** [comment]
so wie ich das verstehe hättest du gerne einen Adapter mit dem du alle Sensoren im ioBroker übewachen möchtest? Und zwar auf

1. Statusänderung
2. Reaktion
3. Online Status

Wie stellst du dir das denn vor? Eine Oberfläche, auf der man jeden Sensor einzeln konfigurieren kann? Oder soll der Adapter alles automatisiert überwachen und man gibt nur vor, was im Falle von "..." passieren soll?

**Comment by klassisch** [comment]
I would start simple as I do it currently with my scripts
* monitoring single data point which were manually selected. These data ponits could be e.g. values of sensors
*  menue to select
  * name of data point to be monitored, that means which (re)triggers the watch dog
  * watchdog time for this data point
  * reaction, e.g. mail , the typical messange systems, message text, etc
  * selectable limitation of the reaction, e.g. max x (e.g. 7) messages per day for this item
  * optional: data point to indicate if the monitored data point is "alive"

To enable bilateral monitoring of two systems an additional oscillator would be helpful. With this method I am currently monitoring ioBroker by HM-CCU and vice versa. very simple and effective.
The oscillator needs
* select the data point used as an output, a simple toggling boolean
* select the cycle time of the oscillator

The selection of the output data points (alive or oscillator output) could be supported by the possibility to create a new data point in a namespace which also could be under 0_userdata (or alias, but I think alias management would be a different sotry).

**Comment by klassisch** [comment]
Btw: such an adapter could also be used to monitor if a window is open for a too long period of time. Also a very popular application.
Watchdogs are indeed very versatilely.

**Comment by oweitman** [comment]
this is the javascript/blockly-adapter or the node-red-adapter, sometimes called as logic machines
you can define individual conditions for reactions
to glue different adapters and functions together

if you want an adapter for a fixed set of conditions and reactions, these have to be defined.
for specialized things these already exists. eg shuttercontrol or alarms and some more.

**Comment by mcm1957** [comment]
should be assigned to hse83 - see #45

**Comment by HSE83** [comment]
I'd like to discuss further ideas in #45 so we have one common request to talk about. I think all the mentioned features should be already there - please double check if you are interested.

**Comment by mcm1957** [comment]
This request might already be covered by an existing adapter:

Watchdog functionality (monitoring devices/last-update intervals) may already be covered by the existing [ioBroker.device-watcher](https://github.com/ioBroker/ioBroker.device-watcher) adapter.

Please check whether it fulfills your needs. / Bitte pruefe, ob dieser Adapter deine Anforderung bereits abdeckt.

**Comment by klassisch** [comment]
Yes, for me it is ok to close this request. I scripted waht I needed. Thanks for cleaning up!
